internal/catalog/internal/controllers/endpoints/controller.go

```diff
@@ -311,8 +311,13 @@ func workloadToEndpoint(svc *pbcatalog.Service, data *workloadData) *pbcatalog.E
 			continue
 		}
 
-		if workloadPort.Protocol != svcPort.Protocol {
-			// workload port mismatch - ignore it
+		// If workload protocol is not specified, we will default to service's protocol.
+		// This is because on some platforms (kubernetes), workload protocol is not always
+		// known, and so we need to inherit from the service instead.
+		if workloadPort.Protocol == pbcatalog.Protocol_PROTOCOL_UNSPECIFIED {
+			workloadPort.Protocol = svcPort.Protocol
```

Comment:
(Sorry for the ignorant question, and I think the answer is "no")

Could there be an unintended side-effect of modifying this in place? Thinking of the case where multiple services with different protocols select the same workload (in practice that's probably a bug, but having the protocol on the workload flap could create strange behavior). It _looks_ like this is always a local service-specific value from a network call, but wanted to double check.

Reply:
Great q! I think we are safe here because even though it's a pointer to the workload port, we never write it back to the resource service(we only use it for reads), and so this value never gets re-used in a way that would cause the actual workload to change.